pandas/core/indexes/base.py

```diff
@@ -1559,6 +1559,18 @@ def _validate_names(
 
         return new_names
 
+    def get_default_index_names(self, df: DataFrame, names: str = None):
+
+        if names is not None and not isinstance(names, str):
+            raise ValueError("Names must be a string")
```

Comment:
Or list/tuple of strings?

Reply:
yeah use `is_hashable` here to be consistent with other cases

Reply:
we likley have tests that hit this..but